pkg/auth/dex.go

```diff
@@ -182,8 +193,13 @@ func (a *DexAppClient) handleDexLogin(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// TODO(BB) Set an app state to make the connection more secure
-	authCodeURL := oauthConfig.AuthCodeURL("APP_STATE")
+	// TODO(BB) Set an app state to make the connection more secure.
+	// Currently a random string is generated but a session state should be built.
+	state, err := generateState()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
```suggestion
		http.Error(w, fmt.Errorf("could not generate state %s", err.Error()), http.StatusInternalServerError)
```

Reply:
Replaced it with a better error message.